File: frontend/src/types/experiment.ts
/**
 * 实验环境类型定义
 */

export type PodPhase = "Running" | "Pending" | "Succeeded" | "Failed" | "Unknown";
export type ComponentHealth = "运行正常" | "需要关注" | "异常";
export type ConnectionStatus = "连接正常" | "连接失败" | "未配置";

export interface PodInfo {
  name: string;
  namespace: string;
  node: string;
  phase: PodPhase;
  restarts: number;
  ready: string; // "1/1" 格式
  ip: string | null;
  created_at: string; // ISO 8601
}

export interface NodeInfo {
  name: string;
  pod_count: number;
  status: "Ready" | "NotReady";
}

export interface ComponentStatus {
  name: string;
  version: string;
  instances: string; // "2/2" 格式
  health: ComponentHealth;
}

export interface ClusterSummary {
  node_count: number;
  namespace_count: number;
  pod_count: number;
  abnormal_pod_count: number;
}

export interface ExperimentEnvironment {
  api_server: string;
  k8s_version: string;
  last_sync: string; // ISO 8601
  connection_status: ConnectionStatus;
  summary: ClusterSummary;
  nodes: NodeInfo[];
  pods: PodInfo[];
  components: ComponentStatus[];
}

File: backend/models/experiment.py
"""实验环境数据模型。"""

from datetime import datetime
from typing import Dict, List, Literal, Optional

from pydantic import BaseModel, Field

PodPhase = Literal["Running", "Pending", "Succeeded", "Failed", "Unknown"]
ComponentHealth = Literal["运行正常", "需要关注", "异常"]


class PodInfo(BaseModel):
    """Pod 信息。"""

    name: str
    namespace: str
    node: str
    phase: PodPhase
    restarts: int
    ready: str  # "1/1" 格式
    ip: Optional[str] = None
    created_at: datetime


class NodeInfo(BaseModel):
    """节点信息。"""

    name: str
    pod_count: int
    status: Literal["Ready", "NotReady"]


class ComponentStatus(BaseModel):
    """资源能力组件状态。"""

    name: str
    version: str
    instances: str  # "2/2" 格式
    health: ComponentHealth


class ClusterSummary(BaseModel):
    """集群统计摘要。"""

    node_count: int
    namespace_count: int
    pod_count: int
    abnormal_pod_count: int


class ExperimentEnvironment(BaseModel):
    """实验环境完整状态。"""

    api_server: str
    k8s_version: str
    last_sync: datetime
    connection_status: Literal["连接正常", "连接失败", "未配置"]
    summary: ClusterSummary
    nodes: List[NodeInfo] = Field(default_factory=list)
    pods: List[PodInfo] = Field(default_factory=list)
    components: List[ComponentStatus] = Field(default_factory=list)
